ddtrace/mocktracer/mocktracer.go

```diff
@@ -50,38 +49,43 @@ type Tracer interface {
 // interface to query the tracer's state.
 func Start() Tracer {
 	t := newMockTracer()
-	internal.SetGlobalTracer(t)
-	internal.Testing = true
+	tracer.SetGlobalTracer(t)
+	tracer.Testing = true
 	return t
 }
 
 type mocktracer struct {
 	sync.RWMutex  // guards below spans
-	finishedSpans []Span
-	openSpans     map[uint64]Span
+	finishedSpans []*Span
+	openSpans     map[uint64]*Span
 }
 
 func newMockTracer() *mocktracer {
```

Comment:
This might be an unpopular opinion, but if this method returned `tracer.Tracer`, then the `var _ tracer.Tracer = (*mocktracer)(nil)` would become redundant.

Reply:
True, but I think we should keep the explicit checks anyway so we don't accidentally stop checking that we implement the interface.